lib/src/types/did_parser.rs

```diff
@@ -285,6 +287,32 @@ mod tests {
         }
     }
 
+    #[test]
+    fn test_parse_hex_c_string_svc() {
+        let decoded_data =
+            hex::decode("6469642f7376632f4d6573736167696e67536572766963650000000000000000")
+                .unwrap();
+        let cstr = CStr::from_bytes_until_nul(decoded_data.as_slice()).unwrap();
+        let parsed = parse_attribute(&cstr.as_ref().to_str().unwrap());
+        assert_eq!(parsed, Ok(Attribute::Service(ServiceType::Messaging)));
+    }
+
+    #[test]
+    fn test_parse_hex_c_string_pub() {
+        let decoded_data =
+            hex::decode("6469642f7075622f456432353531392f766572694b65792f62617365353800").unwrap();
+        let cstr = CStr::from_bytes_until_nul(decoded_data.as_slice()).unwrap();
```

Comment:
We can also statically construct a `DidattributeChanged` struct and call the method directly,

```rust
pub struct DidattributeChangedFilter {
    pub identity: [Address](https://github.com/xmtp/didethresolver/ethereum_types/type.Address.html),
    pub name: [[u8](https://doc.rust-lang.org/1.75.0/std/primitive.u8.html); [32](https://doc.rust-lang.org/1.75.0/std/primitive.array.html)],
    pub value: [Bytes](https://github.com/xmtp/didethresolver/ethers_core/types/bytes/struct.Bytes.html),
    pub valid_to: [U256](https://github.com/xmtp/didethresolver/primitive_types/struct.U256.html),
    pub previous_change: [U256](https://github.com/xmtp/didethresolver/primitive_types/struct.U256.html),
}
```